src/ParametersHandler/include/BipedalLocomotion/ParametersHandler/IParametersHandler.h

```diff
@@ -192,6 +193,12 @@ class IParametersHandler
      */
     virtual std::string toString() const = 0;
 
+    /**
+     * Return the names of all the stored parameters.
+     * @return A set containing the names of all stored parameters.
+     */
+    virtual std::unordered_set<std::string> getParametersNames() const = 0;
```

Comment:
Another, more important, comment would be that ``parameter`` is not the correct wording, as in the handler there may be both parameters and groups. Maybe ``getNames`` is too vague? @GiulioRomualdi what would be your take on this?

Reply:
Right, I started exploring the code of blf two days ago, and I never explored the group feature you mentioned. However, being familiar with YARP I think I know what's the reason of it being there.

I think `getNames` would be perfect if you don't have any term that includes both parameters and groups (that could be _option_ if you want to pick a term).

Reply:
> Right, I started exploring the code of blf two days ago, and I never explored the group feature you mentioned. However, being familiar with YARP I think I know what's the reason of it being there.

Indeed, the reason is to mimic the functionality provided by YARP .ini files and in the future of TOML files (https://toml.io/en/).

Reply:
Otherwise, we can go for the always beautiful wordy solution ``getParametersAndGroupsNames`` :grin:

Reply:
Your call, consider that in Python it would become ever wordier (is this a word?): `get_parameters_and_groups_names` :) I'd tend towards a clean and simple `IParametersHandle::getNames`, I think users understand.

Reply:
Ok for ``getNames`` then :grin:

Reply:
See follow up in https://github.com/dic-iit/bipedal-locomotion-framework/pull/134#discussion_r518428122

Reply:
I like the separated version :+1: